nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/flow-analysis-rules/flow-analysis-rules.effects.ts

```diff
@@ -0,0 +1,536 @@
+                        return this.extensionTypesService
+                            .getImplementingControllerServiceTypes(
+                                // @ts-ignore
+                                descriptor.identifiesControllerService,
+                                descriptor.identifiesControllerServiceBundle
+                            )
+                            .pipe(
+                                take(1),
+                                switchMap((implementingTypesResponse) => {
+                                    // show the create controller service dialog with the types that implemented the interface
+                                    const createServiceDialogReference = this.dialog.open(CreateControllerService, {
+                                        data: {
+                                            controllerServiceTypes: implementingTypesResponse.controllerServiceTypes
+                                        },
+                                        panelClass: 'medium-dialog'
+                                    });
+
+                                    return createServiceDialogReference.componentInstance.createControllerService.pipe(
+                                        take(1),
+                                        switchMap((controllerServiceType) => {
+                                            // typically this sequence would be implemented with ngrx actions, however we are
+                                            // currently in an edit session and we need to return both the value (new service id)
+                                            // and updated property descriptor so the table renders correctly
+                                            return this.managementControllerServiceService
+                                                .createControllerService({
+                                                    revision: {
+                                                        clientId: this.client.getClientId(),
+                                                        version: 0
+                                                    },
+                                                    controllerServiceType: controllerServiceType.type,
+                                                    controllerServiceBundle: controllerServiceType.bundle
+                                                })
+                                                .pipe(
+                                                    take(1),
+                                                    switchMap((createResponse) => {
+                                                        // dispatch an inline create service success action so the new service is in the state
+                                                        this.store.dispatch(
+                                                            ManagementControllerServicesActions.inlineCreateControllerServiceSuccess(
```

Comment:
The controller service state is not loaded when this tab is active. We don't need to dispatch this action here as the services will be loaded when that tab is selected. I took a quick look and it appears I made this mistake earlier in `FlowEffects`. Dispatching `ControllerServicesActions.inlineCreateControllerServiceSuccess` is unnecessary there too. Same for `ReportingTaskEffects`.

This is required in the `ManagementControllerServiceEffects` and `ControllerServiceEffects` because the Edit Controller Service dialog is open over the Controller Service listing. Adding a new service will result in the new service being added to the listing.